### PR TITLE
NXP-25169: add bulk bench scenario

### PR DESCRIPTION
**Pull request by funshodavid**

### CODE REVIEW

**Comment by kevinleturc** [comment]
👍 
we need to find an easy way to open a CoreSession for a given repository and username without having to use both `Framework` and `CoreInstance`.

**Comment by kevinleturc** [comment]
👍

**Comment by kevinleturc** [comment]
:+1:

**Comment by kevinleturc** [comment]
👍

**Comment by pierre-gautier** [comment]
👍

File: nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkService.java [hunk 1]
```diff
@@ -38,14 +38,14 @@
...
 
     /**
-     * Waits for completion of given bulk.
+     * Waits for completion of given bulk command.
      *
-     * @param timeout the time to wait
-     * @param unit the timeout unit
-     * @return {@code true} if bulk completed, or {@code false} if computation has not finished after the timeout
+     * @param commandId the command to wait
+     * @param duration the duration to wait
```
Comment (bdelbosc) [inline nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkService.java]: this is a timeout duration, i.e. the maximum time we want to wait for the command completion

File: nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/StreamBulkProcessor.java [hunk 2]
```diff
@@ -163,55 +166,63 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
...
+                try {
+                    loginContext = Framework.loginAsUser(command.getUsername());
+                    try (CloseableCoreSession session = CoreInstance.openCoreSession(command.getRepository())) {
+                        // scroll documents
+                        Long scrollStartTime = Instant.now().toEpochMilli();
+                        ScrollResult<String> scroll = session.scroll(command.getQuery(), scrollBatchSize,
+                                scrollKeepAliveSeconds);
+                        long documentCount = 0;
+                        long bucketNumber = 0;
+                        while (scroll.hasResults()) {
```
Comment (bdelbosc) [inline nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/StreamBulkProcessor.java]: too much level at some point this should be split in multiple methods

**Pull request merged**